integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanProductWithDownPaymentConfigurationTest.java

```diff
@@ -0,0 +1,312 @@
+        assertEquals(enableDownPayment, loanDetails.getEnableDownPayment());
+        assertEquals(0, loanDetails.getDisbursedAmountPercentageForDownPayment().compareTo(disbursedAmountPercentageForDownPayment));
+
+    }
+
+    @Test
+    public void loanApplicationWithLoanProductWithEnableDownPaymentConfigurationDoesNotChangeWithUpdateProductConfiguration() {
+        // Loan ExternalId
+        String loanExternalIdStr = UUID.randomUUID().toString();
+
+        // Delinquency Bucket
+        final Integer delinquencyBucketId = DelinquencyBucketsHelper.createDelinquencyBucket(requestSpec, responseSpec);
+        final GetDelinquencyBucketsResponse delinquencyBucket = DelinquencyBucketsHelper.getDelinquencyBucket(requestSpec, responseSpec,
+                delinquencyBucketId);
+
+        // down-payment configuration
+        Boolean enableDownPayment = true;
+        BigDecimal disbursedAmountPercentageForDownPayment = BigDecimal.valueOf(12.5);
+
+        final Integer clientId = clientHelper.createClient(ClientHelper.defaultClientCreationRequest()).getClientId().intValue();
+
+        // Loan Product creation with down-payment configuration
+        Integer loanProductId = createLoanProductWithDownPaymentConfiguration(loanTransactionHelper, delinquencyBucketId, enableDownPayment,
+                "12.5");
+
+        final GetLoanProductsProductIdResponse getLoanProductsProductResponse = loanTransactionHelper.getLoanProduct(loanProductId);
+        assertNotNull(getLoanProductsProductResponse);
+        assertEquals(enableDownPayment, getLoanProductsProductResponse.getEnableDownPayment());
+        assertEquals(0, getLoanProductsProductResponse.getDisbursedAmountPercentageForDownPayment()
+                .compareTo(disbursedAmountPercentageForDownPayment));
+
+        final Integer loanId = createLoanAccount(clientId, loanProductId.longValue(), loanExternalIdStr);
+
+        // Retrieve Loan with loanId
+
+        GetLoansLoanIdResponse loanDetails = loanTransactionHelper.getLoanDetails(loanId.longValue());
+
+        // verify down-payment details for Loan
+        assertNotNull(loanDetails);
+        assertEquals(enableDownPayment, loanDetails.getEnableDownPayment());
+        assertEquals(0, loanDetails.getDisbursedAmountPercentageForDownPayment().compareTo(disbursedAmountPercentageForDownPayment));
+
+        // Modify Loan Product to update enable down payment configuration
+        PutLoanProductsProductIdResponse loanProductModifyResponse = updateLoanProduct(loanTransactionHelper,
+                getLoanProductsProductResponse.getId());
+        assertNotNull(loanProductModifyResponse);
+
+        // verify Loan product configuration change
+        GetLoanProductsProductIdResponse getLoanProductsProductResponse_1 = loanTransactionHelper.getLoanProduct(loanProductId);
+        assertNotNull(getLoanProductsProductResponse_1);
+        assertEquals(enableDownPayment, getLoanProductsProductResponse_1.getEnableDownPayment());
```

Comment:
Down payment was already enabled, so it should be disabled in this test case, no?

Reply:
Hi Adam, In this test case i changed disbursedAmount percentage and not enable downpayment config. Earlier for LoanProduct it was 12.5 and loan account was created with 12.5 and then i changed percentage to 25 but loan account has percentage 12.5 with which it was created.

Reply:
Is it not failing on the validation?

I recall if the down payment percentage is passed, but the down payment is not enabled, there is a validation to catch this situations, no?

Reply:
Yes. There is a validation for this situation. I am checking if enableDownPayment is false then downPayment percentage element should not be passed.
I am not setting that to null in db if that is so.